wiki/start/a-simple-game.md

```diff
@@ -5,605 +5,1091 @@ redirect_from:
+If you run the game again, there shouldn't be any difference in the output. That's good! If you don't see the bucket, make sure you adjusted the line indicated above correctly.
+
+### Keyboard
+Now to capturing player input. This is how you detect if a player is pressing keys on the keyboard. This needs to happen in the input method
 
 ```java
-   @Override
-   public void render() {
-      ScreenUtils.clear(0, 0, 0.2f, 1);
+private void input() {
+    if (Gdx.input.isKeyPressed(Keys.RIGHT)) {
+        //todo: Do something when the user presses the right arrow
+    }
+}
+```
+
+This is known as keyboard polling. Every time a frame is drawn, we're going to check if a key is pressed. There is a list of pretty much every conceivable key in Gdx.input.Keys. We want to react to the user pressing the right arrow key.
 
-      ... more to come here ...
-   }
+![keys list](/assets/images/dev/a-simple-game/10.png)
+
+That's great, but what is supposed to happen when the key is pressed? We need to move the coordinates of the bucket sprite.
+
+```java
+private void input() {
+    if (Gdx.input.isKeyPressed(Keys.RIGHT)) {
+        bucketSprite.setX(bucketSprite.getX() + .25f); //Move the bucket right
+    }
+}
+```
+
+The number `.25f` dictates how fast the bucket moves. Adding to the x makes the bucket move to the right. This basically means "set the bucket x to what it currently is right now plus a little bit more". Subtracting from the x makes the bucket move to the left.
+
+An unfortunate side effect of having our logic inside the render method is that our code behaves differently on different hardware. This is because of differences in framerate. More frames per second means more movement per second.
+
+![fps comparison](/assets/images/dev/a-simple-game/11.png)
+
+To counteract this, we need to use delta time. Delta time is the measured time between frames. If we multiply our movement by delta time, the movement will be consistent no matter what hardware we run this game on.
+
+```java
+private void input() {
+    float delta = Gdx.graphics.getDeltaTime(); //retrieve the current delta
+
+    if (Gdx.input.isKeyPressed(Keys.RIGHT)) {
+        bucketSprite.setX(bucketSprite.getX() + .25f * delta); //Move the bucket right
+    }
+}
 ```
 
-The arguments for `ScreenUtils.clear(r, g, b, a)` are the red, green, blue and alpha component of that color, each within the range [0, 1].
+This effectively means that the number we select here is how far the bucket moves in one second. Remember to use delta time whenever you are calculating something that happens over time. Adjust the number to a value that moves the bucket at an acceptable speed like `4f`. Now let's copy the code for movement to the left. Flip the plus to a minus to move to the left.
+
+```java
+private void input() {
+    float delta = Gdx.graphics.getDeltaTime();
+
+    if (Gdx.input.isKeyPressed(Keys.RIGHT)) {
+        bucketSprite.setX(bucketSprite.getX() + 4f * delta); //move the bucket right
```

Comment:
I think it'd be better to move the `4f` magic number up into a `float speed = 4f;` variable.

Reply:
Good suggestion! I've included that in the examples. Thanks.